mypy/checkexpr.py

```diff
@@ -1694,6 +1694,9 @@ def visit_dict_expr(self, e: DictExpr) -> Type:
 
         Translate it into a call to dict(), with provisions for **expr.
         """
+        # if we find a DictExpr inside a context that expected TypedDict,
+        # we try to convert the dictionary into TypedDict
+        # we warn if they are not compatible or if we don't have enough info to verify
```

Comment:
This is not what I wanted. I wanted an explanation that we need to silence one of two errors, and why the code below actually silences the second error. Sorry for misunderstanding.